aws/environment.go

```diff
@@ -66,24 +66,36 @@ func NewEnvironment(ctx *pulumi.Context) (Environment, error) {
 		CommonEnvironment: &commonEnv,
 		Namer:             namer.NewNamer(ctx, awsConfigNamespace),
 		awsConfig:         sdkconfig.New(ctx, awsConfigNamespace),
-		envDefault:        getEnvironmentDefault(commonEnv.InfraEnvironmentName()),
+		envDefault:        getEnvironmentDefault(config.FindEnvironmentName(commonEnv.InfraEnvironmentNames(), awsConfigNamespace)),
 	}
 
 	var err error
-	env.Provider, err = sdkaws.NewProvider(ctx, "aws", &sdkaws.ProviderArgs{
+	env.awsProvider, err = sdkaws.NewProvider(ctx, "aws", &sdkaws.ProviderArgs{
 		Region: pulumi.String(env.Region()),
 		DefaultTags: sdkaws.ProviderDefaultTagsArgs{
 			Tags: commonEnv.ResourcesTags(),
 		},
 	})
+	if err != nil {
+		return Environment{}, err
+	}
 
+	env.awsxProvider, err = sdkawsx.NewProvider(ctx, "awsx", &sdkawsx.ProviderArgs{})
 	if err != nil {
 		return Environment{}, err
 	}
 
 	return env, nil
 }
 
+func (e *Environment) InvokeProviderOption() pulumi.InvokeOption {
+	return pulumi.Provider(e.awsProvider)
+}
+
+func (e *Environment) ResourceProvidersOption() pulumi.ResourceOption {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
What is the difference between a `pulumi.ResourceOption` and a `pulumi.InvokeOption` ? I see `pulumi.Provider(e.Provider)` was changed to one of the two, can you give more details on why you used one and not the other ?

Reply:
It's some Pulumi arcane, it's not documented so I'm not 100% sure, but it's the different types required by the functions:
```
NewXXX creates Resources, hence require `ResourceOption`
Lookup/GetXXX gets some value, hence require `InvokeOption`
```

The `pulumi.Provider(e.Provider)` returns an object that implements both interfaces, and from what I can see in the implementation, `InvokeOption` only tolerate a single provider (unlike `ResourceOption`).

Reply:
Thank you !